Formula/php.rb

```diff
@@ -0,0 +1,343 @@
+class Php < Formula
+  desc "General-purpose scripting language"
+  homepage "https://php.net"
+  url "https://github.com/php/php-src/archive/php-7.1.8.tar.gz"
+  sha256 "f3be4262203fa9db3d126832a1632e9a80e52e313f4230218c392d6a9178c368"
+
+  # javian: not sure about the origin of this so I'll keep it commented for now. Could it have something to do with building extensions ?
+  # So PHP extensions don't report missing symbols
+  # skip_clean "bin", "sbin"
+
+  # TODO
+  # Opcache default config file is included in extension formula, is it needed ?
+  # ldap extension: I have a vague recollection that this caused some issues that were resolved with exluding it. needs to be checked.
+  # Need to add tests for all supported SAPIs
+
+  option "with-imap-uw", "Build PHP IMAP extension"
+  option "with-thread-safety", "Build with thread safety"
+
+  # javian: Not yet checked these options
+  option "with-pdo-oci", "Include Oracle databases (requries ORACLE_HOME to be set)"
+  option "without-ldap", "Build without LDAP support"
+  option "without-unixodbc", "Build without unixODBC support"
+
+  depends_on "autoconf" => :build
+  depends_on "bison" => :build
+  depends_on "re2c" => :build
+  depends_on "imap-uw" => :optional
+  depends_on "libtool" => :run # javian: mcrypt requirement
+  depends_on "aspell"
+  depends_on "argon2"
+  depends_on "curl" if MacOS.version < :lion
+  depends_on "enchant"
+  depends_on "freetds"
+  depends_on "freetype"
+  depends_on "gettext"
+  depends_on "gmp"
+  depends_on "homebrew/apache/httpd24"
+  depends_on "icu4c"
+  depends_on "jpeg"
+  depends_on "libpng"
+  depends_on "mcrypt"
+  depends_on "net-snmp"
+  depends_on "openssl"
+  depends_on "pcre"
+  depends_on "tidy-html5"
+  depends_on "unixodbc"
+  depends_on "webp"
+
+  resource "libpq" do
+    url "https://ftp.postgresql.org/pub/source/v9.6.3/postgresql-9.6.3.tar.bz2"
+    sha256 "1645b3736901f6d854e695a937389e68ff2066ce0cde9d73919d6ab7c995b9c6"
+  end
+
+  # Fixes the pear .lock permissions issue that keeps it from operating correctly.
+  # Thanks mistym & #machomebrew
+  # javian: is this still needed ?
+  skip_clean "lib/php/.lock"
+
+  def install
+    config_path = etc+"php"+version.to_s[0..2]
+    # Not removing all pear.conf and .pearrc files from PHP path results in
+    # the PHP configure not properly setting the pear binary to be installed
+    config_pear = "#{config_path}/pear.conf"
+    user_pear = "#{File.expand_path("~")}/pear.conf"
+    config_pearrc = "#{config_path}/.pearrc"
+    user_pearrc = "#{File.expand_path("~")}/.pearrc"
```

Comment:
This will point to `.brew_home` and not the user's home directory.

Reply:
You cannot get the user's home directory reliably inside a formula at all (also because some users run Homebrew as a different user to the applications).

Reply:
Right, this code should be removed so it won't be an issue that needs a resolution. Thanks for the information.